src/engine/source/builder/asset.cpp

```diff
@@ -154,11 +154,42 @@ Asset::Asset(const json::Json& jsonDefinition,
     // Get stages
     m_stages = base::Chain::create("stages", {});
     auto asOp = m_stages->getPtr<base::Operation>();
+    json::Json definition;
+    definition.setObject();
     for (auto& tuple : objectDefinition)
     {
-        auto stageName = "stage." + std::get<0>(tuple);
+        std::string stageName;
+        base::Expression stageExpression;
+
         auto stageDefinition = std::get<1>(tuple);
-        auto stageExpression = registry->getBuilder(stageName)({stageDefinition}, definitions);
+        auto pos = std::get<0>(tuple).find("parse|");
+
+        if (pos != std::string::npos)
+        {
+            auto key = std::get<0>(tuple).substr(pos + 6);
+            key.erase(std::remove_if(key.begin(), key.end(), ::isspace), key.end());
+            if (stageDefinition.isArray())
+            {
+                json::Json tmp;
+                tmp.setArray();
+                auto arr = stageDefinition.getArray().value();
```

Comment:
This should not go here, the parse is a check not a stage, since it is part of the condition for it to be accepted (the asset):

https://github.com/wazuh/wazuh/blob/c246ba35c0483486a85d93968829c85057b87900/src/engine/source/builder/asset.cpp#L131-L152